crates/webidl/src/util.rs

```diff
@@ -34,35 +38,380 @@ pub fn mdn_doc(class: &str, method: Option<&str>) -> String {
+        }
     }
 }
 
+
 /// Map a webidl const value to the correct wasm-bindgen const value
-pub fn webidl_const_v_to_backend_const_v(v: &webidl::ast::ConstValue) -> backend::ast::ConstValue {
+pub fn webidl_const_v_to_backend_const_v(v: &ConstValue) -> backend::ast::ConstValue {
+    use std::f64::{NEG_INFINITY, INFINITY, NAN};
+    use backend::ast;
+
     match *v {
-        webidl::ast::ConstValue::BooleanLiteral(b) => backend::ast::ConstValue::BooleanLiteral(b),
-        webidl::ast::ConstValue::FloatLiteral(f) => backend::ast::ConstValue::FloatLiteral(f),
-        webidl::ast::ConstValue::SignedIntegerLiteral(i) => backend::ast::ConstValue::SignedIntegerLiteral(i),
-        webidl::ast::ConstValue::UnsignedIntegerLiteral(u) => backend::ast::ConstValue::UnsignedIntegerLiteral(u),
-        webidl::ast::ConstValue::Null => backend::ast::ConstValue::Null,
+        ConstValue::Boolean(b) => ast::ConstValue::BooleanLiteral(b.0),
+        ConstValue::Float(FloatLit::NegInfinity(_)) => {
+            ast::ConstValue::FloatLiteral(NEG_INFINITY)
+        }
+        ConstValue::Float(FloatLit::Infinity(_)) => {
+            ast::ConstValue::FloatLiteral(INFINITY)
+        }
+        ConstValue::Float(FloatLit::NaN(_)) => {
+            ast::ConstValue::FloatLiteral(NAN)
+        }
+        ConstValue::Float(FloatLit::Value(s)) => {
+            ast::ConstValue::FloatLiteral(s.0.parse().unwrap())
+        }
+        ConstValue::Integer(lit) => {
+            let mklit = |orig_text: &str, base: u32, offset: usize| {
+                let (negative, text) = if orig_text.starts_with("-") {
+                    (true, &orig_text[1..])
+                } else {
+                    (false, orig_text)
+                };
+                if text == "0" {
+                    return ast::ConstValue::SignedIntegerLiteral(0)
+                }
+                let text = &text[offset..];
+                match i64::from_str_radix(text, base) {
+                    Ok(n) if negative => ast::ConstValue::SignedIntegerLiteral(-n),
+                    Ok(n) => ast::ConstValue::SignedIntegerLiteral(n),
+                    Err(_) => {
+                        let n = u64::from_str_radix(text, base)
+                            .unwrap_or_else(|_| {
+                                panic!("literal too big: {}", orig_text)
+                            });
+                        if negative {
+                            if n == 1 << 63 {
+                                return ast::ConstValue::SignedIntegerLiteral(i64::min_value())
+                            } else {
+                                panic!("literal too big: {}", orig_text)
+                            }
+                        }
+                        ast::ConstValue::UnsignedIntegerLiteral(n)
+                    }
+                }
+            };
+            match lit {
+                IntegerLit::Hex(h) => mklit(h.0, 16, 2),
+                IntegerLit::Oct(h) => mklit(h.0, 8, 2),
```

Comment:
WebIDL octal literals start with the prefix of `0` rather than `0o`, so the offset here should be 1, not 2.